src/app/core/model/snap-process.ts

```diff
@@ -0,0 +1,134 @@
+import { Proccess, ProcessType } from "./process";
+import { App } from "./app.model";
+import { ElectronService } from "../services";
+import { ChildProcessWithoutNullStreams } from "child_process";
+import { from } from "rxjs";
+import { filter, throwIfEmpty, map } from "rxjs/operators";
+import { AlertController } from '@ionic/angular'
+
+export class SnapProcess implements Proccess {
+
+    onProcessFinishedCallback: Function
+    onProcessUpdatedCallback: Function
+    app: App
+    processType: ProcessType
+    electronService: ElectronService
+    stdout: string[] = []
+    stderr: string[] = []
+    spawn: ChildProcessWithoutNullStreams
+
+    constructor(
+        onProcessFinishedCallback: Function,
+        onProcessUpdatedCallback: Function,
+        app: App,
+        processType: ProcessType,
+        electronService: ElectronService,
+        private alertController: AlertController,
+    ) {
+        this.onProcessFinishedCallback = onProcessFinishedCallback
+        this.onProcessUpdatedCallback = onProcessUpdatedCallback
+        this.app = app
+        this.processType = processType
+        this.electronService = electronService
+    }
+
+    startInstall() {
+        this.presentNOUrlAlert(`Instalando o snap ${this.app.packageName}`)// teste 
+        this.spawn = this.electronService.childProcess.spawn('snap', ['install', this.app.packageName, '-y'])
```

Comment:
Para instalar um snap ainda é necessario saber o confinement e o channel

Reply:
Em relação a essa parte, creio que o problema não tem haver com confinamento ou o canal, pois não estou utilizando ele no formato snap e sim pelo npm start e da mesma forma não estou conseguindo fazer a execução ou instalação de snap e flatpak.

Reply:
O que eu estou a dizer é que, por exemplo, no caso do NodeJS `sudo snap install node --classic` era necessario saber o confinamento do snap para que fosse prossivel a instalaçao

Reply:
bem eu não estou utilizando o NodeJs na forma de snap, porem estou utilizando vscode na forma de snap  
Talvez isso cause alguma interferência, irei baixar tudo manualmente para ver se o problema é esse e for irei atras de encontrar uma solução

Reply:
obrigado pela dica, irei fazer uns teste e te reporto os resultados

Reply:
Ainda nao foi isso que eu quis dizer. A questao é que nessa linha. O que diz `n('snap', ['install', this.app.packageName, '-y'])` formaria um comando de terminal `snap install package-name -y` . O que na maioria dos snaps funciona. Porem, alguns snaps, como no caso do node citado mais acima, necessita colocar `--classic` no comando. Esses casos, bem como os casos que é necessario colocar `--edge`, esse comando nao funcionaria

Reply:
verdade, nesse caso só funcionaria para os que estão no canal estável
Mas uma alternativa é que na hora de instalar o usuário possa escolher por qual canal ele deseja instalar o programa 
exemplo:
estável -> snap install package-name -y
candidato -> snap install package-name ---candidate -y 
borda -> snap install package-name --edge -y
Outra coisa é analisar o tipo de confinamento